#167165046 Setup postgres database

Tunji17 opened the pull request:
### What does this PR do?
- Changes the database from mongo to Postgres
#### Description of Task to be completed?
- Used sequelize ORM to interface with a Postgres database for both test and development
#### How should this be manually tested?

- clone this repo using git clone 
- and run `npm run migrate`

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#167165046](https://www.pivotaltracker.com/story/show/167165046)


#### Screenshots (if appropriate)
![Uploading Screenshot 2019-07-23 at 8.56.02 PM.png…]()

Commit:
chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[start #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[start #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

Commit:
chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

chore(postgres):configure the project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]

mazma1 commented on database/models/index.js:
```diff
@@ -0,0 +1,37 @@
+    db[model.name] = model;
+  });
+
+Object.keys(db).forEach(modelName => {
```
Fixing this won't break anything

mazma1 commented on database/models/index.js:
```diff
@@ -0,0 +1,37 @@
+  .filter(file => {
+    return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
+  })
+  .forEach(file => {
```
Same here, this won't break anything

mazma1 commented on database/models/index.js:
```diff
@@ -0,0 +1,37 @@
+
+fs
+  .readdirSync(__dirname)
+  .filter(file => {
```
Fix this as well

Tunji17 replied:
okay let me work on it

mazma1 commented on database/models/index.js:
```diff
@@ -0,0 +1,37 @@
+
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
```
Fix this

mazma1 commented on database/models/index.js:
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Removing 'use strict' wont break anything

mazma1 commented on database/config/config.js:
```diff
@@ -0,0 +1,15 @@
+
+export const production = {
+  use_env_variable: 'DATABASE_URL'
+};
```
Every file should end with an empty line so fix this

tolulope-od commented on database/models/like.js:
```diff
@@ -0,0 +1,27 @@
+    likedOn: {
+      allowNull: false,
+      type: DataTypes.DATE,
+      defaultValue: DataTypes.literal('CURRENT_TIMESTAMP'),
```
Please delete this. It shouldn't be in the model definition. It's okay in the migrations

tolulope-od commented on database/models/article.js:
```diff
@@ -0,0 +1,83 @@
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+    });
+    Article.hasMany(models.Bookmarks, {
```
This should be `Bookmark`, not `Bookmarks`

tolulope-od commented on database/models/User.js:
```diff
@@ -0,0 +1,95 @@
+      foreignKey: 'userId',
+      otherKey: 'followerId',
+      as: 'followers',
+      through: 'followers',
```
Please change the `followers` to `Followers` so it references the table properly. Thank you

tolulope-od commented on database/models/User.js:
```diff
@@ -0,0 +1,95 @@
+      foreignKey: 'followerId',
+      otherKey: 'userId',
+      as: 'followings',
+      through: 'followers',
```
The same goes for this line; `followers` to `Followers`. Thanks

tolulope-od commented on database/models/User.js:
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable space-before-function-paren */
```
Please change the name of this file, it should start with a lowercase since it doesn't export a class. Thank you

tolulope-od reviewed (commented):
Great job Tunji! You've made some colossal changes to the codebase, please see my comments on some issues I found and kindly implement the fixes I proposed.

Commit:
chore(postgres):configure project to use postgres

setting up postgres and removing the existing mongoose connections

[finishes #167165046]
